packages/components/input/src/input.vue

```diff
@@ -337,17 +337,20 @@ const resizeTextarea = () => {
 const setNativeInputValue = () => {
   const input = _ref.value
   if (!input || input.value === nativeInputValue.value) return
+  if (props.formatter) {
+    input.value = props.formatter(nativeInputValue.value)
+    return
+  }
   input.value = nativeInputValue.value
 }
 
 const handleInput = async (event: Event) => {
   recordCursor()
 
-  let { value } = event.target as TargetElement
-
+  const { value } = event.target as TargetElement
+  let parsedValue
   if (props.formatter) {
-    value = props.parser ? props.parser(value) : value
-    value = props.formatter(value)
```

Comment:
hello, I think just delete this line of code that is ok. Is any other reason for adding the new variable `parsedValue`?